sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -347,20 +338,31 @@ class AccountState(val params: NetworkParams,
 
   override def getCode(address: Array[Byte]): Array[Byte] = using(getView)(_.getCode(address))
 
-  override def getBaseFee: BigInteger = using(getView)(_.getBaseFee)
-
-  override def getBlockGasLimit: BigInteger = using(getView)(_.getBlockGasLimit)
-
   override def validate(tx: SidechainTypes#SCAT): Try[Unit] = Try {
     tx.semanticValidity()
 
     if (tx.isInstanceOf[EthereumTransaction]) {
 
       val ethTx = tx.asInstanceOf[EthereumTransaction]
-      val blockGasPool = new GasPool(getBlockGasLimit)
+      val blockGasLimit = Account.GAS_LIMIT
+      val blockGasPool = new GasPool(BigInteger.valueOf(blockGasLimit))
+      val blockContext = new BlockContext(
+        // use the null address as forger
+        new Array[Byte](32),
+        // TODO: what timestamp do we use here? just use "now()" or can we get the timestamp of the last block?
```

Comment:
i think we need to use `timeProvider: NetworkTimeProvider` method `time()`
So the tx that is out of block is always validated against the current time.

Reply:
What about calculating the timestamp from consensus epoch and slot+1, is that possible? Epoch is already known here, we are just missing the slot.

Reply:
the point with consensus epoch&slot that we don't know the forger block values. It can be `+1` or `+many`.
The consensus protocol we use has a time based slots who may remain empty. From other side, considering again that slots are time based, we may assume that current time is the mostly likely to be used when the block is going to be produced.
Note: we can't use system.time, but timeProvider that considers time shift of the node.